numpy/ma/core.py

```diff
@@ -5869,6 +5869,18 @@ def __getitem__(self, indx):
 
         """
         m = self._mask
+        if m[indx].size > 1:
```

Comment:
Now that I've tested I see your point. 

But what if we get really crazy with a subarray of size 1 of structured dtype:

    >>> a = np.ma.array(np.ones(2, dtype=[('A', 'i4,i4', (1,1,1)), ('B', 'i4')]), 
                        mask=[([[[(0, 0)]]], 0), ([[[(0, 0)]]], 0)])
    >>> a[0]['A']
    masked